uploader/models.py

```diff
@@ -1,3 +1,32 @@
 from django.db import models
 
-# Create your models here.
+
+class SingletonModel(models.Model):
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.pk = 1
+        super(SingletonModel, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        pass
+
+    @classmethod
+    def load(cls):
+        obj, created = cls.objects.get_or_create(pk=1)
+        return obj
+
+
+class FileUploadCounter(SingletonModel):
+    count = models.PositiveIntegerField("Counter", default=0)
```

Comment:
Lets store more information of the upload itself, and later do a count on the number_of_files field. Check comment on the issue, or https://phabricator.wikimedia.org/T241464